src/host/renderData.cpp

```diff
@@ -389,11 +389,50 @@ void RenderData::SelectNewRegion(const COORD coordStart, const COORD coordEnd)
 // - none
 // Return Value:
 // - current selection anchor
-const COORD RenderData::GetSelectionAnchor() const
+const COORD RenderData::GetSelectionAnchor() const noexcept
 {
     return Selection::Instance().GetSelectionAnchor();
 }
 
+// Routine Description:
+// - Gets the current end selection anchor position
+// Arguments:
+// - none
+// Return Value:
+// - current selection anchor
+const COORD RenderData::GetEndSelectionPosition() const noexcept

```

Comment:
how do these handle block selection?

Reply:
In ConHost's selection model, we just draw a block from the start position to the end position. For _line selection_, it does some extra work to figure out what regions outside of the box should be selected.

Reply:
right, but the UIA text range can't know that it's in box or not-box selection mode, right? so when it calls GetText we proceed as though it was line mode?

Reply:
oh dear... This is going to be a lot more intricate then. Because we'll need to create one text range for box selection that is REALLY different from the others.

Reply:
added a TODO #4509